### PR TITLE
bugfix: fix the application was configured to degrade at startup and can't be dynamically switch to upgraded

### PR DESCRIPTION
**Pull request by slievrly**
Signed-off-by: slievrly <slievrly@163.com>

### Ⅰ. Describe what this PR did

bugfix: fix the application was configured to degrade at startup and can't be dynamically switch to upgraded

### Ⅳ. Describe how to verify it

before startup:  service.disableGlobalTransaction=true;
after startup: service.disableGlobalTransaction=true;
Note that you need to use a configuration center that is not of file type. Dynamic configuration does not take effect when using seata-spring-boot-starter && type=file.

### CODE REVIEW

**Comment by xianshi122** [comment]
经过验证，service.disableGlobalTransaction的功能，已经生效
环境：springcloud+nacos+seata，
启动前nacos配置  service.disableGlobalTransaction=true
启动后修改nacos配置  service.disableGlobalTransaction=false，
打印以下日志，初始化  Global Transaction Clients
2020-07-27 15:31:10.343 - INFO -  --- [configListenerOperate_1_1] i.s.s.a.GlobalTransactionScanner        [180]  : Resource Manager is initialized. applicationId[haizol-test-service1] txServiceGroup[haizol_tx_group]
2020-07-27 15:31:10.344 - INFO -  --- [configListenerOperate_1_1] i.s.s.a.GlobalTransactionScanner        [184]  : Global Transaction Clients are initialized. 
操作事务相关业务，注册RM 全局事务已触发
image

File: spring/src/main/java/io/seata/spring/annotation/GlobalTransactionScanner.java [hunk 1]
```diff
@@ -78,8 +80,9 @@
     private final String applicationId;
     private final String txServiceGroup;
     private final int mode;
-    private final boolean disableGlobalTransaction = ConfigurationFactory.getInstance().getBoolean(
+    private boolean disableGlobalTransaction = ConfigurationFactory.getInstance().getBoolean(
```
Comment (wangliang181230) [inline spring/src/main/java/io/seata/spring/annotation/GlobalTransactionScanner.java:83]: volatile ?

**Comment by a364176773** [comment]
resolve code conflicts

File: spring/src/main/java/io/seata/spring/annotation/GlobalTransactionScanner.java (hunk 1 above)
Comment (slievrly) [reply spring/src/main/java/io/seata/spring/annotation/GlobalTransactionScanner.java:83]: done

File: spring/src/main/java/io/seata/spring/tcc/TccActionInterceptor.java [hunk 2]
```diff
@@ -32,17 +36,22 @@
...
  * @author zhangsen
  */
-public class TccActionInterceptor implements MethodInterceptor {
+public class TccActionInterceptor implements MethodInterceptor, ConfigurationChangeListener {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(TccActionInterceptor.class);
 
     private ActionInterceptorHandler actionInterceptorHandler = new ActionInterceptorHandler();
 
+    private boolean disable = ConfigurationFactory.getInstance().getBoolean(ConfigurationKeys.DISABLE_GLOBAL_TRANSACTION,
```
[the remark this answers is not in the document]
Comment (slievrly) [reply spring/src/main/java/io/seata/spring/tcc/TccActionInterceptor.java:52]: for the upgrade and downgrade functions，needs to be explicitly printed in the log.

[the remark this answers is not in the document]
Comment (wangliang181230) [reply spring/src/main/java/io/seata/spring/tcc/TccActionInterceptor.java:52]: volatile

**Review by wangliang181230 — APPROVED** [review]
LGTM

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by l81893521 — APPROVED** [review]
LGTM

**Pull request merged**